File: services/cs/TrinityService/services/util/DocumentationServiceApi.cs
using System.Collections.Generic;
using System.Net.Http;
using System.ServiceModel;
using System.ServiceModel.Web;
using com.trafigura.services.meta;

namespace com.trafigura.services.util
{
    [ServiceContract]
    public interface DocumentationServiceApi
    {
        [OperationContract]
        [WebGet(UriTemplate = "Files/{filename}")]
        HttpResponseMessage File(string filename);

        [OperationContract]
        [WebGet(UriTemplate = "Meta/{path}")]
        WebService MetaData(string path);

        [OperationContract]
        [WebGet(UriTemplate = "Services")]
        List<WebService> AllMetaData();
    }
}

File: services/src/com/trafigura/services/DocumentationServiceApi.scala
package com.trafigura.services

import javax.ws.rs._
import javax.ws.rs.core._
import meta.WebService


@Path("/Doc")
trait DocumentationServiceApi {
  @Path("Files/{filename}") @GET
  def file(@PathParam("filename") filename: String): Response

  @Path("Forms/{serviceUri}") @GET @Produces(Array("text/html"))
  def forms(@PathParam("serviceUri") serviceUri: String): String

  @Path("Meta/{serviceUri}") @GET @Produces(Array("application/json"))
  def metaData(@PathParam("serviceUri") serviceUri: String): WebService

  @Path("Services") @GET @Produces(Array("application/json"))
  def allMetaData: List[WebService]
}